### PR TITLE
[mqtt] Read settings from ENV

### PR DESCRIPTION
**Pull request by dmolokanov**
Read settings from ENV.
Use pem cert and pkey pair instead of pkcs12
Remove broker.json config. Use defaults instead.

### CODE REVIEW

File: edge-hub/watchdog/src/main.rs [hunk 1]
```diff
@@ -43,7 +43,7 @@ fn main() -> Result<()> {
     let broker_handle = match run(
         "MQTT Broker",
         "/usr/local/bin/mqttd",
-        vec!["-c".to_string(), "/app/mqttd/broker.json".to_string()],
+        vec!["-c".to_string()],
```
Comment (vadim-kovalyov) [inline edge-hub/watchdog/src/main.rs:46]: remove `-c`

File: mqtt/mqtt-broker/src/transport.rs [hunk 2]
```diff
@@ -57,20 +57,33 @@ impl Transport {
...
             acceptor.cert_store_mut().add_cert(ca)?;
         }
 
-        acceptor.set_verify(SslVerifyMode::PEER);
+        // set options to support some clients
+        acceptor.set_options(
+            SslOptions::NO_SESSION_RESUMPTION_ON_RENEGOTIATION | SslOptions::NO_TICKET,
+        );
+
+        // request client certificate for verification but disabel certificate verification
```
Comment (vadim-kovalyov) [inline mqtt/mqtt-broker/src/transport.rs:82]:
```suggestion
        // request client certificate for verification but disable certificate verification
```

File: mqtt/mqtt-edgehub/src/settings.rs [hunk 3]
```diff
@@ -194,4 +234,41 @@ mod tests {
...
+    #[test]
+    fn from_file_overrides_settings_from_env() {
+        it_overrides_settings_from_env(|| Settings::from_file("config/default.json"));
+    }
+
+    fn it_overrides_settings_from_env<F>(make_settings: F)
+    where
+        F: FnOnce() -> Result<Settings, ConfigError>,
+    {
+        env::set_var("LISTENER__TCP__ADDRESS", "10.0.0.1:1883");
```
Comment (vadim-kovalyov) [inline mqtt/mqtt-edgehub/src/settings.rs:252]: so, that's the format for env vars?

**Comment by dmolokanov** [comment]
Reverting config changes

**Pull request closed without merging**